common/djangoapps/enrollment/views.py

```diff
@@ -469,6 +472,14 @@ def get(self, request):
                     ).format(username=username)
                 }
             )
+        if username == request.user.username or self.has_api_key_permissions(request):
+            return Response(enrollment_data)
+        filtered_data = []
+        for enrollment in enrollment_data:
```

Comment:
Are instructors/admins expected to have access too?  I have the same question for organizational users.

Depending on the answer, you may want to use `courseware.access.has_access` instead to provide the expected access.